### PR TITLE
Invalid view changes

### PR DESCRIPTION
**Pull request by bzawisto**
This PR fixes (partially) https://github.com/Zilliqa/zq2/issues/1142 and https://github.com/Zilliqa/zq2/issues/1105.

Txns are pulled one by one from mempool and included in the block based on their effective gas usage (not the max limit specified in txn).

### CODE REVIEW

File: zilliqa/src/pool.rs [hunk 1]
```diff
@@ -124,11 +124,6 @@ impl TransactionPool {
                 continue;
             };
 
-            // If we've popped a nonced transaction, that may have made a subsequent one valid
-            if let Some(next) = tx_index.next().and_then(|idx| self.transactions.get(&idx)) {
-                self.ready.push(next.into());
-            }
-
```
Comment (JamesHinshelwood) [inline zilliqa/src/pool.rs:131]: @bzawisto Why did we remove this? I'm going to re-add it since I can't imagine this code working without it

Comment (JamesHinshelwood) [reply zilliqa/src/pool.rs:131]: Oh I see that you rely on the push in `mark_executed` instead. Will add a clarifying comment.

File: zilliqa/tests/it/main.rs [hunk 2]
```diff
@@ -263,7 +263,7 @@ impl Network {
             consensus: ConsensusConfig {
                 genesis_deposits: genesis_deposits.clone(),
                 is_main: send_to_parent.is_none(),
-                consensus_timeout: Duration::from_secs(1),
+                consensus_timeout: Duration::from_secs(5),
```
Comment (JamesHinshelwood) [inline zilliqa/tests/it/main.rs:266]: @bzawisto Why these changes? Will leave them in but an explanation might be useful.

**Review by JamesHinshelwood — APPROVED** [review]

**Review by DrZoltanFazekas — APPROVED** [review]
